Spelling

jsoref opened the pull request:
## What do these changes do?

Fixes misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/GraphScope/actions/runs/7936794632/attempts/1#summary-21672807577

The action will report that the changes in this PR would make it happy: https://github.com/jsoref/GraphScope/actions/runs/7936795177/attempts/1#summary-21672809066

## Related issue number

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/alibaba/GraphScope?pullRequest=3547) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/alibaba/GraphScope?pullRequest=3547) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alibaba/GraphScope?pullRequest=3547) it.</sub>

jsoref commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -25,7 +25,7 @@ If applicable, add screenshots to help explain your problem.
 **Environment (please complete the following information):**
  - GraphScope version: [e.g., v0.1, v0.2.2, master]
- - OS: [e.g. MacOS, Linux]
+ - OS: [e.g. macOS, Linux]
```
brand

jsoref commented on .github/workflows/gaia.yml:
```diff
@@ -88,12 +88,12 @@ jobs:
         . ${HOME}/.graphscope_env
         git clone -b master --single-branch --depth=1 https://github.com/7br/gstest.git /tmp/gstest
-        cd ${GITHUB_WORKSPACE}/interactive_engine/compiler && ./ir_exprimental_ci.sh
+        cd ${GITHUB_WORKSPACE}/interactive_engine/compiler && ./ir_experimental_ci.sh
```
I've tried to rename related files...

jsoref commented on .github/free-disk-space.sh:
```diff
@@ -22,7 +22,7 @@
 # This script frees up 28 GB of disk space by deleting unneeded packages and 
 # large directories.
-# The Flink end to end tests download and generate more than 17 GB of files,
+# The following end to end tests download and generate more than 17 GB of files,
```
It took me a while to decide that this was probably the intended word

jsoref reviewed (commented):
I'll try to add notes about the rest of the changes later... It'll take me a while.

Note: I'm more than happy to split things off / exclude items. I just need input.

sighingnow commented:
Closing as we cannot accept such a monster PR for spelling.

jsoref commented:
I'm happy to split it up in any manner you request

sighingnow commented:
> I'm happy to split it up in any manner you request

We currently only consider typos in documentation and comments. The typo in variables in sources and YAML/JSON configurations requires further case-by-case investigation.

Thanks!

jsoref commented:
I've split off most of the comments changes. If that manages to be merged, I'll see about making a PR for doc changes, and we can go from there.

Note that it's purely a heuristic, so there may be some comment fixes that I dropped.

sighingnow commented:
> ## What do these changes do?
> Fixes misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).
> 
> The misspellings have been reported at https://github.com/jsoref/GraphScope/actions/runs/7936794632/attempts/1#summary-21672807577
> 
> The action will report that the changes in this PR would make it happy: https://github.com/jsoref/GraphScope/actions/runs/7936795177/attempts/1#summary-21672809066
> 
> ## Related issue number

Hi @jsoref, thanks a lot for those spelling fixes. From the link you pasted in the description, I cannot only find the list of unrecognized words. I'm curious about how the commits (applying suggestions) are generated?

I would like to add your check-spelling action to this repo as well as our other repos as well. Could you please show me some wiki/docs about it?

Thanks!

sighingnow commented:
> Depending on the project and my current work-mode, I take different approaches to correcting misspellings. Sometimes I'll manually correct things, sometimes I let chrome suggest a correction, and sometimes I let Google Sheets suggest corrections.

I see. Thanks again for you effort!
